Core/MIPS/x86/X64IRAsm.cpp

```diff
@@ -192,21 +210,26 @@ void X64JitBackend::GenerateFixedCode(MIPSState *mipsState) {
 #elif PPSSPP_ARCH(AMD64)
 			MOV(32, R(SCRATCH1), MComplex(MEMBASEREG, SCRATCH1, SCALE_1, 0));
 #endif
-			MOV(32, R(EDX), R(SCRATCH1));
 			_assert_msg_(MIPS_JITBLOCK_MASK == 0xFF000000, "Hardcoded assumption of emuhack mask");
-			SHR(32, R(EDX), Imm8(24));
-			CMP(32, R(EDX), Imm8(MIPS_EMUHACK_OPCODE >> 24));
+			if (cpu_info.bBMI2) {
+				RORX(32, EDX, R(SCRATCH1), 24);
```

Comment:
Hm, I'm confused, if the other bits are not zero, with RORX they'l reappear on the other side, and confuse the CMP?

Reply:
But it's an 8 bit EAX CMP, so it won't be confused.

-[Unknown]

Reply:
Ahah, that's what I missed.